exporter/dataprepperexporter/dataprepper.go

```diff
@@ -0,0 +1,210 @@
+package dataprepperexporter
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	"go.uber.org/zap"
+	"google.golang.org/genproto/googleapis/rpc/status"
+	"google.golang.org/protobuf/proto"
+)
+
+type exporter struct {
+	// Input configuration.
+	config     *Config
+	client     *http.Client
+	tracesURL  string
+	logger     *zap.Logger
+	hasAWSAuth bool
+	hasSigV4   bool
+	header     string
+	region     string
+	signer     *v4.Signer
+}
+
+const (
+	headerRetryAfter         = "Retry-After"
+	maxHTTPResponseReadBytes = 64 * 1024
+	headerDataPrepper        = "x-amz-es-dp-internal"
+	service                  = "es"
+)
+
+// Crete new exporter.
+func newExporter(cfg config.Exporter, logger *zap.Logger) (*exporter, error) {
+	oCfg := cfg.(*Config)
+
+	if oCfg.Endpoint != "" {
+		_, err := url.Parse(oCfg.Endpoint)
+		if err != nil {
+			return nil, errors.New("endpoint must be a valid URL")
+		}
+	}
+
+	hasAWSAuth := HasAWSAuth(*oCfg)
+	hasSigV4 := HasSigV4(oCfg.AWSAuthConfig)
+	header := ""
+	if hasAWSAuth {
+		header = oCfg.AWSAuthConfig.PipelineName
+	}
+
+	client, err := oCfg.HTTPClientSettings.ToClient()
+	if err != nil {
+		return nil, err
+	}
+
+	var signer *v4.Signer
+	if hasSigV4 {
+		res := v4.NewSigner(getCredsFromConfig(oCfg.AWSAuthConfig.SigV4Config))
+		v4.WithUnsignedPayload(res)
```

Comment:
let's add a comment here about why we're setting this flag: our auth model requires the setting of the 'amz-content-sha256' header and because of constraints in the aws-sdk-go library this won't be set unless this flag is explicitly set.

Reply:
Fixed in the new commit